core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -586,15 +585,15 @@ private TempBlockMeta createBlockMetaInternal(long sessionId, long blockId,
       if (newBlock) {
         checkTempBlockIdAvailable(blockId);
       }
-      StorageDirView dirView =
-          mAllocator.allocateBlockWithView(sessionId, initialBlockSize, location, getUpdatedView());
-      if (dirView == null) {
+      StorageDir dir = mAllocator.allocateBlockWithTierInfo(sessionId, initialBlockSize, location,
+          mMetaManager.getTiers(), mMetaManager.getAliasToTiersMap());
```

Comment:
Do we also want to create another `BlockMetadataManagerView` to provide read only access to the tiers and directories?

Reply:
+1. i like that better than exposing the tiers and alias to tiers maps